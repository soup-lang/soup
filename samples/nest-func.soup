func xor(a, b)
    if a
        return !b
    return b

func leap(y)
    func mod4()
        return y % 4 = 0
    func mod100()
        return y % 100 = 0
    func mod400()
        return y % 400 = 0
    m4: mod4()
    m100: mod100()
    m400: mod400()
    return xor(m4, xor(m100, m400))

console.log(leap(4))
console.log(leap(10))
console.log(leap(20))
console.log(leap(100))
console.log(leap(200))
console.log(leap(400))

func deepNest(x)
    z: 5
    func inner1Level(y)
        func inner2Level()
            return x * y
        return inner2Level()
    return inner1Level(z)

console.log(deepNest(2))
