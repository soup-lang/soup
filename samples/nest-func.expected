true
false
true
false
false
true
10
